src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -816,6 +821,46 @@ private DeleteOrganisationResponse deleteOrganisationEntity(Organisation organis
         return deleteOrganisationResponse;
     }
 
+    @Override
+    @Transactional
+    public DeleteUserResponse deleteUserForOrganisation(List<String> emails) {
+        var deleteOrganisationResponse = new DeleteOrganisationResponse();
+        if (emails.isEmpty()) {
+            throw new InvalidRequest("Please provide both email addresses");
+        }
+
+        Set<String> userIdsToBeDeleted = emails.stream().map(
+            email -> {
+                Set<String> userIds = new HashSet<>();
+                Optional.ofNullable(professionalUserRepository
+                        .findByEmailAddress(RefDataUtil.removeAllSpaces(email)))
+                    .ifPresentOrElse(professionalUser -> {
+                        userIds.add(professionalUser.getUserIdentifier());
+                        userAttributeRepository.deleteByProfessionalUserId(professionalUser.getId());
+                        professionalUserRepository.delete(professionalUser);
```

Comment:
lines 839 and 840:
waiting on IO inside a stream will block all other requests from all other spring-boot threads (all streams in JVM use a separate thread pool called CFJP - CommonForkJoinPool), both calls should be moved outside stream processing into a CompletableFuture call or else for-each loop.